tests/test_hierarchy.cc

```diff
@@ -209,6 +209,6 @@ BOOST_AUTO_TEST_CASE(hierarchy_2d)
     res[i + 1] = rel_residual;
   }
 
-  pcout << "Convergence rate: " << res[n_cycles] / res[n_cycles - 1]
-        << std::endl;
+  pcout << "Convergence rate: " << std::fixed << std::setprecision(2)
```

Comment:
You should removed the `std::scientific` from line 198 if you want `std::fixed`

Reply:
I want scientific for residuals, but fixed for the rate.